test/governance/extensions/GovernorTimelockAccess.test.js

```diff
@@ -1,383 +1,363 @@
+const { GovernorHelper } = require('../../helpers/governance');
+const { bigint: Enums } = require('../../helpers/enums');
+const { bigint: time } = require('../../helpers/time');
+const { max } = require('../../helpers/math');
 const { selector } = require('../../helpers/methods');
 const { hashOperation } = require('../../helpers/access-manager');
 
-const AccessManager = artifacts.require('$AccessManager');
-const Governor = artifacts.require('$GovernorTimelockAccessMock');
-const AccessManagedTarget = artifacts.require('$AccessManagedTarget');
-const Ownable = artifacts.require('$Ownable');
-
 const TOKENS = [
-  { Token: artifacts.require('$ERC20Votes'), mode: 'blocknumber' },
-  { Token: artifacts.require('$ERC20VotesTimestampMock'), mode: 'timestamp' },
+  { Token: '$ERC20Votes', mode: 'blocknumber' },
+  { Token: '$ERC20VotesTimestampMock', mode: 'timestamp' },
 ];
 
-contract('GovernorTimelockAccess', function (accounts) {
-  const [admin, voter1, voter2, voter3, voter4, other] = accounts;
+const name = 'OZ-Governor';
+const version = 1n;
+const tokenName = 'MockToken';
+const tokenSymbol = 'MTKN';
+const tokenSupply = ethers.parseEther('100');
+const votingDelay = 4n;
+const votingPeriod = 16n;
+const value = ethers.parseEther('1');
+
+async function fixture() {
+  const [admin, voter1, voter2, voter3, voter4, other] = await ethers.getSigners();
+
+  const manager = await ethers.deployContract('$AccessManager', [admin]);
+  const receiver = await ethers.deployContract('$AccessManagedTarget', [manager]);
+
+  const configs = {};
+  for (const { Token, mode } of TOKENS) {
+    const token = await ethers.deployContract(Token, [tokenName, tokenSymbol, version]);
+    const mock = await ethers.deployContract('$GovernorTimelockAccessMock', [
+      name,
+      votingDelay,
+      votingPeriod,
+      0n,
+      manager,
+      0n,
+      token,
+      0n,
+    ]);
+
+    await admin.sendTransaction({ to: mock, value });
+    await token.$_mint(admin, tokenSupply);
+
+    const helper = new GovernorHelper(mock, mode);
+    await helper.connect(admin).delegate({ token, to: voter1, value: ethers.parseEther('10') });
+    await helper.connect(admin).delegate({ token, to: voter2, value: ethers.parseEther('7') });
+    await helper.connect(admin).delegate({ token, to: voter3, value: ethers.parseEther('5') });
+    await helper.connect(admin).delegate({ token, to: voter4, value: ethers.parseEther('2') });
+
+    configs[Token] = { token, mock, helper };
+  }
+
+  return { admin, voter1, voter2, voter3, voter4, other, manager, receiver, configs };
+}
 
-  const name = 'OZ-Governor';
-  const version = '1';
-  const tokenName = 'MockToken';
-  const tokenSymbol = 'MTKN';
-  const tokenSupply = web3.utils.toWei('100');
-  const votingDelay = web3.utils.toBN(4);
-  const votingPeriod = web3.utils.toBN(16);
-  const value = web3.utils.toWei('1');
+function prepareOperation({ sender, target, value = 0n, data = '0x' }) {
```

Comment:
The default should be at least 4 bytes long, otherwise the `data.slice(0, 10)` for the selector would get a different value, isn't it?

```suggestion
function prepareOperation({ sender, target, value = 0n, data = '0x00000000' }) {
```

Reply:
I think a better option is to keep `data = '0x'` by default but then do 
```
selector: data.slice(0, 10).padEnd(10,'0'),
```

Reply:
Right, that works!